rpc/dial.go

```diff
@@ -259,9 +279,11 @@ func dial(
 	if dOpts.debug {
 		logger.Debugw("trying direct", "address", address)
 	}
-	conn, cached, err = dialDirectGRPC(ctx, address, dOpts, logger)
-	if err != nil {
-		return nil, false, err
+
+	var directErr error
+	conn, cached, directErr = dialDirectGRPC(ctx, address, dOpts, logger)
+	if directErr != nil {
+		return nil, false, multierr.Combine(directErr, nonFatalErr)
```

Comment:
flyby: include error(s) from initial dial attempt if we fail to dial directly as a fallback